pycuda/reduction.py

```diff
@@ -267,7 +268,9 @@ def __call__(self, *args, **kwargs):
                 macroblock_size = block_count*self.block_size
                 seq_count = (sz + macroblock_size - 1) // macroblock_size
 
-            if block_count == 1:
+            if block_count == 1 and out is not None:
+                result = out
```

Comment:
You should check that it has the right size and dtype.

Reply:
I added a check for the dtype. Why the size - only one element is used ?

Reply:
At the very least, there should be a check that the array is non-empty (i.e. size > 0). Additionally, I think it helps set expectations: If someone passes a bigger-than-single-entry array, they might expect all entries to get filled. Plus, it's easy to get and pass a single-entry view of an array if that's what's desired.

Reply:
OK, I added the size test